Add German store IDs

muffix opened the pull request:
Adds a readme file with German Apple retail store IDs

muffix committed:
Add German store IDs

Adds a readme file with German Apple retail store IDs

muffix commented:
This is going to require support querying the German store, similar to what #2 does for Australia and Canada. Happy to add it once #2 has been merged.
The SKUs in Germany appear to be different, too.

worthbak commented:
nice - appreciate you pulling those store IDs together! #2 was just merged if you wanna take a look at adding Germany to the list of configurable countries?

muffix committed:
Parametrise the SKU list and favourites

We can dynamically create SKU lists and favourites. The SKUs all have
the same format of 5 letters, a country code, and end in `/A`. We don't
need to spell them out for each country. It's enough to configure the
code once.

muffix committed:
Add retail stores sorted by city

muffix commented:
Updated the PR. 

The SKUs all follow the same format: First 5 letters, then a country code, then `/A`. There's no need to list all of them for each country in the `constants.js` file. Instead, we can just have them be generated for the country in question. That should make it significantly easier to add support for more countries in the future. All we need to add is the path for the store and the country code of the SKU (`/de` and `D` for Germany, `/au` and `X` for Australia, etc).

muffix commented:
Also added all Apple Retail stores sorted by city.

muffix commented:
Wrong button, apologies. I didn't mean to close this. Can you have a look again @worthbak?

worthbak commented on index.js:
```diff
@@ -1,43 +1,66 @@
 
+const countryConfig = COUNTRIES[country];
+
+let storePath = countryConfig["storePath"];
+let skuList = skusForCountry(countryConfig["skuCode"]);
+let favorites = favouritesForCountry(countryConfig["skuCode"]);
```
just curious - why not use dot-syntax here? ex:
```js
let storePath = countryConfig.storePath;
let skuList = skusForCountry(countryConfig.skuCode);
let favorites = favouritesForCountry(countryConfig.skuCode);
```

muffix replied:
Mostly because I'm not a regular JS developer. 😅

worthbak replied:
hahah that's mostly why i asked - I'm used to working in Swift, so I just assume others know more about JS than i do 😆

worthbak reviewed (approved):
excellent work - thanks @muffix!! very cool insight into the model number variances between countries - I hadn't caught that difference. this will make it dramatically easier to fix issues with other countries (UK is next). merging 👍
